datafusion/core/src/physical_plan/joins/hash_join_utils.rs

```diff
@@ -0,0 +1,598 @@
+///     side: JoinSide::Left, // child side
+/// }
+/// ```
+pub fn map_origin_col_to_filter_col(
+    filter: &JoinFilter,
+    schema: &SchemaRef,
+    side: &JoinSide,
+) -> Result<HashMap<Column, Column>> {
+    let filter_schema = filter.schema();
+    let mut col_to_col_map: HashMap<Column, Column> = HashMap::new();
+    for (filter_schema_index, index) in filter.column_indices().iter().enumerate() {
+        if index.side.eq(side) {
+            // Get the main field from column index:
+            let main_field = schema.field(index.index);
+            // Create a column expression:
+            let main_col = Column::new_with_schema(main_field.name(), schema.as_ref())?;
+            // Since the order of by filter.column_indices() is the same with
+            // that of intermediate schema fields, we can get the column directly.
+            let filter_field = filter_schema.field(filter_schema_index);
+            let filter_col = Column::new(filter_field.name(), filter_schema_index);
+            // Insert mapping:
+            col_to_col_map.insert(main_col, filter_col);
+        }
+    }
+    Ok(col_to_col_map)
+}
+
+/// This function analyzes [PhysicalSortExpr] graphs with respect to monotonicity
+/// (sorting) properties. This is necessary since monotonically increasing and/or
+/// decreasing expressions are required when using join filter expressions for
+/// data pruning purposes.
+///
+/// The method works as follows:
+/// 1. Maps the original columns to the filter columns using the `map_origin_col_to_filter_col` function.
+/// 2. Collects all columns in the sort expression using the `PhysicalExprColumnCollector` visitor.
+/// 3. Checks if all columns are included in the `column_mapping_information` map.
+/// 4. If all columns are included, the sort expression is converted into a filter expression using the `transform_up` and `convert_filter_columns` functions.
+/// 5. Searches the converted filter expression in the filter expression using the `check_filter_expr_contains_sort_information`.
+/// 6. If an exact match is encountered, returns the converted filter expression as `Some(Arc<dyn PhysicalExpr>)`.
+/// 7. If all columns are not included or the exact match is not encountered, returns `None`.
+///
+/// Examples:
+/// Consider the filter expression "a + b > c + 10 AND a + b < c + 100".
+/// 1. If the expression "a@ + d@" is sorted, it will not be accepted since the "d@" column is not part of the filter.
+/// 2. If the expression "d@" is sorted, it will not be accepted since the "d@" column is not part of the filter.
+/// 3. If the expression "a@ + b@ + c@" is sorted, all columns are represented in the filter expression. However,
+///    there is no exact match, so this expression does not indicate pruning.
+pub fn convert_sort_expr_with_filter_schema(
+    side: &JoinSide,
+    filter: &JoinFilter,
+    schema: &SchemaRef,
+    sort_expr: &PhysicalSortExpr,
+) -> Result<Option<Arc<dyn PhysicalExpr>>> {
```

Comment:
I think you can also put `EquivalenceProperties` into the consideration, you can do this in the following PR.

Reply:
Noted.